cli/command/idresolver/idresolver.go

```diff
@@ -36,8 +36,8 @@ func (r *IDResolver) get(ctx context.Context, t any, id string) (string, error)
 			// TODO(thaJeztah): should error-handling be more specific, or is it ok to ignore any error?
 			return id, nil //nolint:nilerr // ignore nil-error being returned, as this is a best-effort.
 		}
-		if node.Spec.Annotations.Name != "" {
-			return node.Spec.Annotations.Name, nil
+		if node.Spec.Name != "" {
+			return node.Spec.Name, nil
```

Comment:
I'm still not sure if we want to enable the `QF1008` ("Omit embedded fields from selector expression") check. While in some cases it's OK to skip the implicit embedded reference, there's many cases where explicit may be preferred over implicit, and being explicit can (for some) help discover situations where a "wrapper" struct is passed, but only an embedded struct is needed.